kuma/users/views.py

```diff
@@ -935,3 +935,37 @@ def stripe_hooks(request):
         )
 
     return HttpResponse()
+
+
+def _send_payment_received_email(payment_intent, locale):
+    user = get_user_model().objects.get(stripe_customer_id=payment_intent.customer)
+    subscription_info = _retrieve_and_synchronize_subscription_info(user)
+    locale = locale or settings.WIKI_DEFAULT_LANGUAGE
+    context = {
+        "payment_date": datetime.fromtimestamp(payment_intent.created),
+        "next_payment_date": subscription_info["next_payment_at"],
+        "invoice_number": payment_intent.number,
+        "cost": settings.CONTRIBUTION_AMOUNT_USD,
+        "credit_card_brand": subscription_info["brand"],
+        "manage_subscription_url": absolutify(reverse("recurring_payment_management")),
+        "faq_url": absolutify(reverse("payments_index")),
+        "contact_email": settings.CONTRIBUTION_SUPPORT_EMAIL,
+    }
+    with translation.override(locale):
+        subject = render_email("users/email/payment_received/subject.ltxt", context)
+        # Email subject *must not* contain newlines
+        subject = "".join(subject.splitlines())
+        plain = render_email("users/email/payment_received/plain.ltxt", context)
+        send_mail_retrying(
+            subject,
+            plain,
+            settings.DEFAULT_FROM_EMAIL,
+            [user.email],
+            attachment={
+                "name": os.path.basename(urlparse(payment_intent.invoice_pdf).path),
+                "bytes": requests_retry_session()
```

Comment:
nit: Can you move this to *above* the `send_mail_retrying` otherwise, if the SMTP retrying actually happens, you'll be downloading the invoice over and over. 

Also, although it's unlikely, I think you should always assert something about HTTP responses. Because they can fail in two completely different ways `RequestError` and `not response.ok`.

So, perhaps something like this:

```python
download = requests_retry_session().get(payment_intent.invoice_pdf)
download.raise_for_status()
send_mail_retrying(
   ...,
   attachment={
       "bytes": download.content,
        ...
  }
```

Reply:
Great point wrt raising the error, I changed it accordingly.

Wrt retrying-triggering-multiple-downloads, I don't think that would've happened, as the bytes were read before it's being passed to the mail call, right?

Reply:
> I don't think that would've happened, as the bytes were read before it's being passed to the mail call, right?

You are absolutely right! It's not a function, it's the result of the function.